### DISCUSSION TITLE
Error when setting bandwidth to 40MHz

### DISCUSSION THREAD

**Discussion by clement-joly-kissfly** [svpcom/wfb-ng#209] [Questions]
I'm trying to play with configuration parameters to increase the range.
I was trying to set the channel bandwidth to 40MHz (up from 20MHz) but I'm getting an error message.

Here is what I've done: 
I've added the following lines to /etc/wifibroadcast.cfg:
On drone side:
```
[drone_video]
bandwidth = 40     # bandwidth 20 or 40 MHz
```
On gs side:
```
[gs_video]
# Radio settings for RX
bandwidth = 40     # bandwidth 20 or 40 MHz
```
When I start the wfb service and check its status (with ```sudo systemctl status wifibroadcast@drone```) I get the following error message:
```
● wifibroadcast@drone.service - WFB profile drone
     Loaded: loaded (/lib/systemd/system/wifibroadcast@.service; disabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Wed 2022-03-09 15:47:04 PST; 698ms ago
    Process: 1810 ExecStart=/usr/bin/wfb-server drone ${WFB_NICS} (code=exited, status=1/FAILURE)
   Main PID: 1810 (code=exited, status=1/FAILURE)

Mar 09 15:47:04 odroid systemd[1]: wifibroadcast@drone.service: Main process exited, code=exited, status=1/FAILURE
Mar 09 15:47:04 odroid systemd[1]: wifibroadcast@drone.service: Failed with result 'exit-code'.
```
If I run ```journalctl -e``` I get:
<pre>
Mar 09 15:49:33 odroid kernel: IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
Mar 09 15:49:38 odroid systemd[1]: wifibroadcast@drone.service: Scheduled restart job, restart counter is at 30.
Mar 09 15:49:38 odroid systemd[1]: Stopped WFB profile drone.
Mar 09 15:49:38 odroid systemd[1]: Started WFB profile drone.
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] Log opened.
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] # iw reg set BO
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] # ifconfig wlan0 down
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] # iw dev wlan0 set monitor otherbss
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] # ifconfig wlan0 up
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] kernel reports: (extension) channel is disabled
Mar 09 15:49:39 odroid wfb-server[2065]:         command failed: Invalid argument (-22)
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] <b>Stopping reactor due to fatal error: RC 234: iw dev wlan0 set channel 10 HT40+</b>
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] Main loop terminated.
Mar 09 15:49:39 odroid wfb-server[2065]: 2022-03-09 15:49:39-0800 [-] Exiting with code 1
Mar 09 15:49:40 odroid systemd[1]: wifibroadcast@drone.service: Main process exited, code=exited, status=1/FAILURE
Mar 09 15:49:40 odroid systemd[1]: wifibroadcast@drone.service: Failed with result 'exit-code'.
Mar 09 15:49:40 odroid kernel: IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
Mar 09 15:49:45 odroid systemd[1]: wifibroadcast@drone.service: Scheduled restart job, restart counter is at 31.
Mar 09 15:49:45 odroid systemd[1]: Stopped WFB profile drone.
Mar 09 15:49:45 odroid systemd[1]: Started WFB profile drone.
</pre>
So it looks like setting the channel width to 40 fails. I was able to get the same error message by running ```iw dev wlan0 set channel 10 HT40+``` from the command line.
On the other hand when I run ```iw dev wlan0 set channel 10 HT20``` or ```iw dev wlan0 set channel 10 HT40-``` I'm not getting any errors.
What am I doing wrong?

**Comment by svpcom** [comment]
Not all channel numbers are valid for 40MHz. See https://en.wikipedia.org/wiki/List_of_WLAN_channels

**Reply by clement-joly-kissfly** [reply]
Thanks Vasily!
